src/formats/PDB.cpp

```diff
@@ -85,6 +88,9 @@ void PDBFormat::read(Frame& frame) {
         case Record::CONECT:
             read_CONECT(frame, line);
             break;
+        case Record::MODEL:
+            ++step_;
+            break;
         case Record::END:
             goto end; // We have read a frame!
```

Comment:
Just reading everything again, and this is bugging me: what happens if we have a PDB file like this:

```
MODEL 1
...
ENDMDL
END
MODEL 2
...
ENDMDL
END
```

Because how `Record::END` is matched, does this means that we will read two frames, the real one and an empty one ?

Reply:
The thing is, there should be only 1 `END` record. I was actually surprised to see one of the examples (`wat.pdb` I think?) that used the `END` record more like as an `ENDMDL`. It seems that is common practice in some chemistry fields? I've always worked with biomolecules so I wouldn't know.
I think that, if Chemfiles supports PDBs with more than 1 `END` record, fine. But if it doesn't, then that's fine too, because that's poor formatting. But, again, I'm biased by my own personal experience on my field.

Reply:
Yes, this is an issue with file formats in chemistry: there might be a standard (and we are already lucky when there is one), and then there are the implementations and different way of doing of all the existing codes.

My usual point of view on that is that chemfiles should write files conforming to the standard as much as possible, and accept as many files as possible when reading, even non-conforming ones if possible. I think I'll try to fix this! Give me some time and I'll add a commit on this PR.

Reply:
869a0629b7c4b9b432708f2978aa3055bb789e73 should do the trick here. Tell me what you think.

FYI, I also changed code in the forward function, because file.readline() changes the position of the cursor in the file stream, so we need to store it and restore it afterwards using `tellg` and `seekg`.

Reply:
> FYI, I also changed code in the forward function, because file.readline() changes the position of the cursor in the file stream, so we need to store it and restore it afterwards using tellg and seekg.

Yeap, sorry for introducing a bug. I thought of that and then forgot to fix it.

So, correct me if I'm wrong, but now files like this one:
```
MODEL 1    // well formatted file
...
ENDMDL
MODEL 2
...
```
or this one:

```
ATOM    1 ....     // a typical chemistry PDB
...
END
ATOM    1 ....
...
```
or this one:
```
MODEL 656      // a horrendous quimera of the above
...
ENDMDL
END
MODEL 123
...
```
would result in the same `Trajectory` object. Right?
I think is great to have such a robust library. Do you want me to add warnings for the above 2 cases? Or would they become too annoying when working with PDB chemistry files?

Reply:
>would result in the same Trajectory object. Right?

Yes, you are right.

> Do you want me to add warnings for the above 2 cases? Or would they become too annoying when working with PDB chemistry files?

We could add a warning for the last case, which is really a weird one. For the second case it might become very annoying quickly.

Reply:
Ah, if its too rare then I wouldn't worry about it. The second case was the one that I cared about the most, but you're right, chemistry people would hate such a warning. I think we're done!